experiment/cherrypicker/server.go

```diff
@@ -0,0 +1,197 @@
+		s.Logger.Info(resp)
+		return s.GithubClient.CreateComment(org, repo, num, plugins.FormatResponseRaw(ic.Issue.Body, ic.Issue.HTMLURL, commentAuthor, resp))
+	}
+
+	// Only members should be able to do cherry-picks.
+	ok, err := s.GithubClient.IsMember(org, commentAuthor)
+	if err != nil {
+		return err
+	}
+	if !ok {
+		resp := fmt.Sprintf("only %s org members may request cherry picks", org)
+		s.Logger.Info(resp)
+		return s.GithubClient.CreateComment(org, repo, num, plugins.FormatResponseRaw(ic.Issue.Body, ic.Issue.HTMLURL, commentAuthor, resp))
+	}
+
+	// Cherry-pick only merged PRs.
+	if !pr.Merged {
+		resp := "cannot cherry-pick an unmerged PR"
+		s.Logger.Info(resp)
+		return s.GithubClient.CreateComment(org, repo, num, plugins.FormatResponseRaw(ic.Issue.Body, ic.Issue.HTMLURL, commentAuthor, resp))
+	}
+
+	// Clone the repo, checkout the target branch.
+	startClone := time.Now()
+	r, err := s.GitClient.Clone(ic.Repo.FullName)
+	if err != nil {
+		return err
+	}
+	defer func() {
+		if err := r.Clean(); err != nil {
+			s.Logger.WithError(err).Error("Error cleaning up repo.")
+		}
+	}()
+	if err := r.Checkout(targetBranch); err != nil {
+		return err
+	}
+	s.Logger.WithField("duration", time.Since(startClone)).Info("Cloned and checked out target branch.")
+
+	// Fetch the patch from Github
+	rsp, err := http.Get(fmt.Sprintf("https://patch-diff.githubusercontent.com/raw/%s/%s/pull/%d.patch", org, repo, num))
+	if err != nil {
+		return err
+	}
+	defer rsp.Body.Close()
+	if rsp.StatusCode < 200 || rsp.StatusCode > 299 {
+		return fmt.Errorf("response status was not 2XX: %s", rsp.Status)
+	}
+	localPath := fmt.Sprintf("/tmp/%d-%s.patch", num, uuid.NewV1().String())
+	out, err := os.Create(localPath)
+	if err != nil {
+		return err
+	}
+	defer out.Close()
+	if _, err := io.Copy(out, rsp.Body); err != nil {
+		return err
+	}
+
+	if err := r.Config("user.name", "cherrypicker"); err != nil {
+		return err
+	}
+	if err := r.Config("user.email", "cherrypicker@localhost"); err != nil {
+		return err
+	}
+
+	// Apply the patch.
+	if err := r.Apply(localPath); err != nil {
+		resp := fmt.Sprintf("PR %d failed to apply on top of branch %q: %v", num, targetBranch, err)
+		return s.GithubClient.CreateComment(org, repo, num, plugins.FormatResponseRaw(ic.Issue.Body, ic.Issue.HTMLURL, commentAuthor, resp))
+	}
+
+	// Push a new branch with the applied patch in bot's fork.
+	newBranch := fmt.Sprintf("cherry-pick-%d-to-%s", num, targetBranch)
+	if err := r.CheckoutNewBranch(newBranch); err != nil {
+		return err
+	}
+	if err := r.Push(s.BotName, s.BotCredentials, repo, newBranch); err != nil {
+		return err
+	}
+
+	// Open a PR in Github.
+	title := fmt.Sprintf("Automated cherry-pick of #%d on %s", num, targetBranch)
+	// TODO: Assign release manager if someone in the org wears that hat.
+	body := fmt.Sprintf("/assign %s", commentAuthor)
```

Comment:
Can get the original release note and add to the new PR body?

Reply:
It'd be good to also add more info (original PR number for xref, original title for clarity) as what we do in the manual cherry-pick process.
See the example: https://github.com/kubernetes/kubernetes/pull/53885#issue-265234809

Reply:
Yeah, the body can be extended to include more info or even better use a template or something.